From b77a750bed05c6ffc7adf2b3936169f7e1c85e11 Mon Sep 17 00:00:00 2001
From: GengFei <fei.geng@windriver.com>
Date: Tue, 16 Mar 2010 15:00:19 +0800
Subject: [PATCH] tracing/markers: make markers select tracepoints

Impact: make markers select tracepoints

commit 91f73f90d97fa67effbb49e0a79c50cf26dfe324 upstream

    Sometimes it happens that KConfig dependencies are not handled
    like in the following scenario:

    - config A
       bool

    - config B
       bool
       depends on A

    - config C
       bool
       select B

    If one selects C, then it will select B without checking its
    dependency to A, if A hasn't been selected elsewhere, it will
    result in a build failure.

    This is what happens on the following build error:

     kernel/built-in.o: In function `marker_update_probe_range':
     (.text+0x52f64): undefined reference to `tracepoint_probe_register_noupdate'
     kernel/built-in.o: In function `marker_update_probe_range':
     (.text+0x52f74): undefined reference to `tracepoint_probe_unregister_noupdate'
     kernel/built-in.o: In function `marker_update_probe_range':
     (.text+0x52fb9): undefined reference to `tracepoint_probe_unregister_noupdate'
     kernel/built-in.o: In function `marker_update_probes':
     marker.c:(.text+0x530ba): undefined reference to `tracepoint_probe_update_all'

    CONFIG_KVM_TRACE will select CONFIG_MARKER, but the latter
    depends on CONFIG_TRACEPOINTS which will not be selected.

    Reported-by: Ingo Molnar <mingo@elte.hu>
    Signed-off-by: Frederic Weisbecker <fweisbec@gmail.com>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>
    Intergrated-by: GengFei <fei.geng@windriver.com>
---
 init/Kconfig |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/init/Kconfig b/init/Kconfig
index a2cef77..1c5b397 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -868,7 +868,7 @@ config TRACEPOINTS
 
 config MARKERS
 	bool "Activate markers"
-	depends on TRACEPOINTS
+	select TRACEPOINTS
 	help
 	  Place an empty function call at each marker site. Can be
 	  dynamically changed for a probe function.
-- 
1.6.0.4

